src/extension/serializer.ts

```diff
@@ -1003,3 +1011,105 @@ export class GrpcSerializer extends SerializerBase {
     return this.notebookDataCache.get(cacheId)
   }
 }
+// there is w/ grpc the : serializer, server, runner,
+export class ConnectSerializer extends SerializerBase {
+  protected readonly ready: ReadyPromise
+  log: ReturnType<typeof getLogger>
+  client: PromiseClient<typeof ParserService>
+  serializerServiceUrl: string
+
+  constructor(
+    protected context: ExtensionContext,
+    serializerAddress: string,
+    kernel: Kernel,
+  ) {
+    super(context, kernel)
+    this.log = getLogger('ConnectSerializer')
+    this.serializerServiceUrl = serializerAddress
+    this.client = this.createSerializerClient()
+    this.ready = new Promise((resolve) => {
+      resolve()
+    })
+  }
+
+  createSerializerClient = () => {
+    this.log.info(`Serializer: Client pointed to: ${this.serializerServiceUrl}`)
+    return createPromiseClient(
+      ParserService,
+      createConnectTransport({
```

Comment:
For manual testing purposes, as discussed in other places I think it will be helpful if you can point this at a fully functional implementation of the serializer server so you can verify this is working.
The Runme server has a working implementation which uses the gRPC protocol.
With the connect client you can optionally select what protocol to use. Right now you are using the Connect Transport. However, you can optionally use gRPC by calling the function to create a gRPC transport. That would allow you to test that your implementation is actually working.

Reply:
after our discussion about the tls issues and disabling them, I have discovered the source of the issue.  for the issue & solution see `fake_server/client2.ts` and `fake_server/readme.md` .   I subsequently incorporated a tls connection into `serializer.ts` also